src/v/cluster/archival/tests/ntp_archiver_test.cc

```diff
@@ -2536,3 +2536,120 @@ FIXTURE_TEST(test_ntp_archiver_upload_loop_blocked, archiver_fixture) {
                && probe.value().get_num_paused_archivers() > 0;
     });
 }
+
+// NOLINTNEXTLINE
+FIXTURE_TEST(
+  test_ntp_archiver_upload_loop_not_blocked_if_allow_gaps_set,
+  archiver_fixture) {
+    scoped_config cfg;
+    cfg.get("cloud_storage_disable_upload_loop_for_tests").set_value(true);
+    // Allow-gaps is set to true so the
+    cfg.get("cloud_storage_enable_remote_allow_gaps").set_value(true);
+
+    std::vector<segment_desc> segments = {
```

Comment:
nitpick: might factor out the code that sets up the gappy manifest and all that. the mechanics of how it's done don't feel germane to the test case.